### PR TITLE
[curlpp] Fix target "curlpp" link "ZLIB::ZLIB" error

### PR DESCRIPTION
**Pull request by LilyWangL**
**Describe the pull request**
Add zlib to default feature, fix curlpp build failed with this error:
```
CMake Error at D:/Git/vs2019.prod/scripts/buildsystems/vcpkg.cmake:217 (_add_library):
Target "curlpp" links to target "ZLIB::ZLIB" but the target was not found.
Perhaps a find_package() call is missing for an IMPORTED target, or an
ALIAS target is missing?
Call Stack (most recent call first):
CMakeLists.txt:89 (add_library)
```
- What does your PR fix? Fixes issue #10525 
- All features test pass.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

File: ports/curl/0010_fix_ZlibFeature.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b0355d1..a386e9d 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -352,7 +352,7 @@ set(USE_OPENSSL OFF)
+ if(CMAKE_USE_OPENSSL)
+   find_package(OpenSSL REQUIRED)
+   set(SSL_ENABLED ON)
+-  set(USE_OPENSSL ON)
++  set(USE_OPENSSL 1)
```
Comment (dan-shaw) [inline ports/curl/0010_fix_ZlibFeature.patch]: I'm not sure what this patch does: both ON and 1 are `True`

**Review by dan-shaw — APPROVED** [review]

File: ports/curl/0010_fix_ZlibFeature.patch (hunk 1 above)
Comment (JackBoosY) [reply ports/curl/0010_fix_ZlibFeature.patch]: @dan-shaw After testing, setting `ON` does not work.

**Comment by Neumann-A** [comment]
Please change this PR to use the fix discussed in #10715. The problem is not with curl but with curlpp not setting `cmake_minimum_required()` before the first `project()` call.

**Pull request merged**